pkg/proxy/util/utils.go

```diff
@@ -251,6 +251,29 @@ func GetNodeAddresses(cidrs []string, nw NetworkInterfacer) (sets.String, error)
 	return uniqueAddressList, nil
 }
 
+type IpSetValidator interface {
+	IpIsValidForSet(ip net.IP) bool
+}
+
+func AddressSet(validator IpSetValidator, addrs []net.Addr) sets.String {
```

Comment:
we should add a comment explaining what this function does

Reply:
I re-worked the function. The validator is now a function, which is much better. I did not know that you can pass a method as a function in go (you can't do that in C++).